src/main/java/io/supertokens/jwt/JWTSigningKey.java

```diff
@@ -142,7 +131,17 @@ public JWTSigningKeyInfo getOrCreateAndGetKeyForAlgorithm(SupportedAlgorithms al
 
                     // If no key was found create a new one
                     if (keyInfo == null) {
-                        keyInfo = generateAndSetKeyInStorage(sqlStorage, con, algorithm);
+                        while (true) {
+                            try {
+                                keyInfo = generateKeyForAlgorithm(algorithm);
+                                sqlStorage.setJWTSigningKey_Transaction(con, keyInfo);
+                                return keyInfo;
```

Comment:
No return.. We need to break since we are doing commit below.

Reply:
Given this bug, i wonder why your tests pass. I think the transaction logic commits on its own if we don't call the commit? Or is it that your tests don't test this case?

Reply:
The transaction logic auto commits but fixed it regardless